core/lib/Cardano/Address/Style/Byron.hs

```diff
@@ -297,21 +298,31 @@ deriveAddressPrivateKey acctK =
 -- structured JSON that gives information about an address.
 --
 -- @since 2.0.0
-inspectByronAddress :: MonadThrow m => Address -> m Json.Value
-inspectByronAddress addr = do
+inspectByronAddress :: forall m. MonadThrow m => Maybe XPub -> Address -> m Json.Value
+inspectByronAddress mRootPub addr = do
     payload <- either (throwM . BrDeserialiseError) pure
         $ CBOR.deserialiseCbor CBOR.decodeAddressPayload bytes
+
     (root, attrs) <- either (throwM . BrDeserialiseError) pure
         $ CBOR.deserialiseCbor decodePayload payload
-    path  <- maybe (throwM BrMissingExpectedDerivationPath) pure
-        $ find ((== 1) . fst) attrs
+
+    path <- do
+        attr <- maybe (throwM BrMissingExpectedDerivationPath) pure
+            (find ((== 1) . fst) attrs)
+        case mRootPub of
+            Nothing ->
+                pure $ toJSON $ T.unpack $ T.decodeUtf8 $ encode EBase16 $ snd attr
+            Just rootPub ->
+                decryptPath rootPub attr
+
     ntwrk <- either (throwM . BrDeserialiseError) pure
         $ CBOR.deserialiseCbor CBOR.decodeProtocolMagicAttr payload
+
     pure $ Json.object
         [ "address_style"   .= Json.String "Byron"
         , "stake_reference" .= Json.String "none"
         , "address_root"    .= T.unpack (T.decodeUtf8 $ encode EBase16 root)
-        , "derivation_path" .= T.unpack (T.decodeUtf8 $ encode EBase16 $ snd path)
+        , "derivation_path" .= path
```

Comment:
I think I've been wondering this before... but there's no specification for the json output? Now the derivation path could be encrypted or decrypted. I think we should document that somewhere?

Reply:
I started to think the same. It'd be good to have a json-schema documenting the output indeed (and test checking against that schema?)

Reply:
Another idea would be to have a separate field for the decrypted path?